Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx

```diff
@@ -56,165 +57,30 @@ using namespace o2::framework;
-        }
-      } catch (const std::exception& e) {
-        std::cout << "Failed to get padId: " << e.what() << std::endl;
-        return;
-      }
-
-      Digit::Time time;
-      time.sampaTime = sc.sampaTime;
-      time.bunchCrossing = sc.bunchCrossing;
-      time.orbit = orbit;
-
-      digits.emplace_back(o2::mch::Digit(deId, padId, digitadc, time));
-
-      if (mPrint)
-        std::cout << "DIGIT STORED:\nADC " << digits.back().getADC() << " DE# " << digits.back().getDetID() << " PadId " << digits.back().getPadID() << " time " << digits.back().getTime().sampaTime << std::endl;
-      ++ndigits;
-    };
-
-    const auto patchPage = [&](gsl::span<const std::byte> rdhBuffer) {
-      auto rdhPtr = const_cast<void*>(reinterpret_cast<const void*>(rdhBuffer.data()));
-      mNrdhs++;
-      auto cruId = o2::raw::RDHUtils::getCRUID(rdhPtr);
-      auto endpoint = o2::raw::RDHUtils::getEndPointID(rdhPtr);
-      o2::raw::RDHUtils::setFEEID(rdhPtr, cruId * 2 + endpoint);
-      orbit = o2::raw::RDHUtils::getHeartBeatOrbit(rdhPtr);
-      if (mPrint) {
-        std::cout << mNrdhs << "--\n";
-        o2::raw::RDHUtils::printRDH(rdhPtr);
-      }
-    };
-
-    if (!mDecoder) {
-      mDecoder = mFee2Solar ? o2::mch::raw::createPageDecoder(page, channelHandler, mFee2Solar)
-                            : o2::mch::raw::createPageDecoder(page, channelHandler);
-    }
-
-    patchPage(page);
-    mDecoder(page);
-  }
-
- private:
-  std::string readFileContent(const std::string& filename)
-  {
-    std::string content;
-    std::string s;
-    std::ifstream in(filename);
-    while (std::getline(in, s)) {
-      content += s;
-      content += " ";
-    }
-    return content;
-  }
-
-  void initElec2DetMapper(const std::string& filename)
-  {
-    if (filename.empty()) {
-      mElec2Det = createElec2DetMapper<ElectronicMapperGenerated>();
-    } else {
-      ElectronicMapperString::sFecMap = readFileContent(filename);
-      mElec2Det = createElec2DetMapper<ElectronicMapperString>();
-    }
-  }
-
-  void initFee2SolarMapper(const std::string& filename)
-  {
-    if (filename.empty()) {
-      mFee2Solar = createFeeLink2SolarMapper<ElectronicMapperGenerated>();
-    } else {
-      ElectronicMapperString::sCruMap = readFileContent(filename);
-      mFee2Solar = createFeeLink2SolarMapper<ElectronicMapperString>();
-    }
-  }
-
  public:
   //_________________________________________________________________________________________________
   void init(framework::InitContext& ic)
   {
-    mNrdhs = 0;
-
-    for (int i = 0; i < 64; i++) {
-      for (int j = 0; j < 64; j++) {
-        if (refManu2ds_st345[j] != i) {
-          continue;
-        }
-        refDs2manu_st345[i] = j;
-        break;
-      }
-    }
-
-    mDs2manu = ic.options().get<bool>("ds2manu");
+    auto mDs2manu = ic.options().get<bool>("ds2manu");
     mPrint = ic.options().get<bool>("print");
 
+    mDecoder.setDs2manu(mDs2manu);
```

Comment:
see my comment above about regular constructor vs bunch of set methods

Reply:
See above.